pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1042,6 +1042,15 @@ func (s *SlirpPodInterface) configureTapDevice() error {
 	return nil
 }
 
-func generateTapDeviceName(vmiUUID string) string {
-	return fmt.Sprintf("tap%s", vmiUUID[0:tapDeviceNameMaxSuffixLength])
+func generateTapDeviceName(vmiUUID string, podInterfaceName string) string {
+	// we must make sure that interfaces connecting multus networks generate
+	// different tap device names.
+	suffix := ""
+	maxTapDeviceSuffixLength := tapDeviceNameMaxSuffixLength
+	if strings.HasPrefix(podInterfaceName, "net") {
+		ifaceIndex := strings.ReplaceAll(podInterfaceName, "net", "")
+		suffix = fmt.Sprintf("_%s", ifaceIndex)
+		maxTapDeviceSuffixLength = tapDeviceNameMaxSuffixLength - len(suffix)
+	}
+	return fmt.Sprintf("tap%s%s", vmiUUID[0:maxTapDeviceSuffixLength], suffix)
```

Comment:
Why the vmUUID is needed? Why not 'tap0' for the default one and 'tap' + suffix for the other?

Reply:
Oh! You're right :) I wanted to prevent name collision on concurrent create vmi requests, but somehow lost track that the interface is created **in** the virt-launcher, not in the node.

My bad, I'll remove it.